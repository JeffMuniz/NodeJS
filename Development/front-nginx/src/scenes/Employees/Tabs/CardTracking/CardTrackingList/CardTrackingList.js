import React from "react";
import { arrayOf, shape } from "prop-types";
import DateManager from "src/modules/DateManager/DateManager";
import { Table, Thead, Tbody, Th, Tr, Td } from "src/styles/table.styles";

import { productTypes, dateHourFormats } from "@enums";

const CardTrackingList = ({ vouchers }) => (
  <Table>
    <Thead>
      <Tr>
        <Th width="80px">Dt. solicitação</Th>
        <Th width="220px">Nome</Th>
        <Th centered width="50px">
          Produto
        </Th>
        <Th width="140px">Nº do Cartão</Th>
        <Th width="100px">Status do cartão</Th>
        <Th>Endereço</Th>
        <Th>Status da entrega</Th>
      </Tr>
    </Thead>
    <Tbody>
      {vouchers.length &&
        vouchers.map((card, index) => (
          <Tr key={index}>
            <Td>
              {DateManager.utc(card.issueDate).format(
                dateHourFormats.shortDateSlash,
              )}
            </Td>
            <Td>{card.printedName}</Td>
            <Td centered>{productTypes[card.idProduct].abbreviation}</Td>
            <Td>{card.cardNumber}</Td>
            <Td>
              <strong>{card.nomeStatus}</strong>
            </Td>
            <Td>{card.address}</Td>
            <Td>{card.trackStatus}</Td>
          </Tr>
        ))}
    </Tbody>
  </Table>
);

CardTrackingList.propTypes = {
  vouchers: arrayOf(shape({})),
};

CardTrackingList.defaultProps = {
  vouchers: [],
};

export default CardTrackingList;
