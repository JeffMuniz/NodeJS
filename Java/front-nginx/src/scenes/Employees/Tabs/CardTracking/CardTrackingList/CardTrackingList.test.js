import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import CardTrackingList from "./CardTrackingList";

describe("CardTrackingList", () => {
  it("Should render CardTrackingList", () => {
    // given
    const renderer = new ShallowRenderer();
    const vouchers = [
      {
        id: "1",
        printedName: "TESTE",
        idProduct: "1",
        cardNumber: "1111xxxxxxxx2222",
        issueDate: "2018-09-19T14:38:56.707Z",
      },
    ];

    // when
    renderer.render(<CardTrackingList vouchers={vouchers} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
