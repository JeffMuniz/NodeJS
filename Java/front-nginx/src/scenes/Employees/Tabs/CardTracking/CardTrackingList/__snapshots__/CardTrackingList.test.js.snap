// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTrackingList Should render CardTrackingList 1`] = `
<styled.table>
  <styled.thead>
    <styled.tr>
      <styled.th
        width="80px"
      >
        Dt. solicitação
      </styled.th>
      <styled.th
        width="220px"
      >
        Nome
      </styled.th>
      <styled.th
        centered={true}
        width="50px"
      >
        Produto
      </styled.th>
      <styled.th
        width="140px"
      >
        Nº do Cartão
      </styled.th>
      <styled.th
        width="100px"
      >
        Status do cartão
      </styled.th>
      <styled.th>
        Endereço
      </styled.th>
      <styled.th>
        Status da entrega
      </styled.th>
    </styled.tr>
  </styled.thead>
  <styled.tbody>
    <styled.tr>
      <styled.td>
        19/09/18
      </styled.td>
      <styled.td>
        TESTE
      </styled.td>
      <styled.td
        centered={true}
      >
        VA
      </styled.td>
      <styled.td>
        1111xxxxxxxx2222
      </styled.td>
      <styled.td>
        <strong />
      </styled.td>
      <styled.td />
      <styled.td />
    </styled.tr>
  </styled.tbody>
</styled.table>
`;
